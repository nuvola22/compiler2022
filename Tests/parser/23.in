function max(num1, num2: integer): integer;
var result: integer;
begin
   if num1 = num2 then result := num1
   else result := num2;
   max := result;
end;

function a(): Integer;
begin
    a := 30;
end;

begin 
    writeln(10, ' ', 30);
    writeln(a);
end.