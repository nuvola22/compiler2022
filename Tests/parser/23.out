      program
         max
            parameters:
               num1
               integer
               num2
               integer
            declarations:
               variable
                  result
                  integer
            type:
               integer
            begin
               if
                  exp:
                     =
                        num1
                        num2
                  body:
                     :=
                        result
                        num1
                  else:
                     :=
                        result
                        num2
               :=
                  max
                  result
         a
            parameters:
            declarations:
            type:
               Integer
            begin
               :=
                  a
                  30
         begin
            call
               writeln
               10
               ' '
               30
            call
               writeln
               a
